test/e2e/prometheus_test.go

```diff
@@ -0,0 +1,297 @@
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("Error retrieving metrics from %s. response: %v", url, err)
+	}
+
+	return string(body)
+}
+
+// Locate Antrea pods from the specified component, retrieve and validate that supplied metric exists
+func testPrometheusMetricsOnPods(t *testing.T, data *TestData, component string, metrics []string) {
+	token := getMonitoringAuthToken(t, data)
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=antrea,component=" + component,
+	}
+	pods, err := data.clientset.CoreV1().Pods(antreaNamespace).List(listOptions)
+	if err != nil {
+		t.Fatalf("Error fetching agent pods: %v", err)
+	}
+
+	var hostIP = ""
+	var hostPort int32 = 0
+	var parser expfmt.TextParser
+
+	// Find pods' connectivity, check for metrics existence on each of them
+	for _, pod := range pods.Items {
+		hostIP = pod.Status.HostIP
+
+		for _, container := range pod.Spec.Containers {
+			for _, port := range container.Ports {
+				hostPort = port.HostPort
+				t.Logf("Found %s %d", hostIP, hostPort)
+				respBody := getMetricsFromApiServer(t, fmt.Sprintf("https://%s:%d/metrics", hostIP, hostPort), token)
+
+				parsed, err := parser.TextToMetricFamilies(strings.NewReader(respBody))
+				if err != nil {
+					t.Fatalf("Parsing Prometheus metrics failed with: %v", err)
+				}
+
+				// Create a map of all the metrics which were found on the server
+				testMap := make(map[string]bool)
+				for _, mf := range parsed {
+					testMap[mf.GetName()] = true
+				}
+
+				// Validate that all the required metrics exist in the server's output
+				for _, metric := range metrics {
+					if !testMap[metric] {
+						t.Fatalf("Metric %s not found on %s:%d", metric, hostIP, hostPort)
+					}
+				}
+			}
+		}
+	}
+}
+
+// Retrieve Prometheus endpoint from K8S
+func getPrometheusEndpoint(t *testing.T, data *TestData) (string, int32) {
+	pods, err := data.clientset.CoreV1().Pods("monitoring").List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching monitoring pods: %v", err)
+	}
+
+	// Find hostIP by querying the Prometheus pod
+	var hostIP = ""
+	for _, pod := range pods.Items {
+		hostIP = pod.Status.HostIP
+	}
+
+	// Find nodePort by querying the Prometheus service
+	services, err := data.clientset.CoreV1().Services("monitoring").List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Error fetching monitoring services: %v", err)
+	}
+
+	var nodePort int32 = 0
+	for _, service := range services.Items {
+		for _, port := range service.Spec.Ports {
+			nodePort = port.NodePort
+		}
+	}
+	if hostIP == "" || nodePort == 0 {
+		t.Fatal("Failed to locate Prometheus endpoint")
+	}
+
+	return hostIP, nodePort
+}
+
+// Validate that metrics are returned from Prometheus client on the Antrea Controller
+// Check for specific metric as the entire set will be tested with Prometheus server
+func TestPrometheusMetricsOnController(t *testing.T) {
```

Comment:
I think the "best practice" in Go is for function comments to start with the name of the function. I think we have been trying to follow that even for test code, at least for the main test functions themselves.